WordPress/wordpress-playground
# Allow blueprint validation to work with CSP

## Motivation for the change, related issues

[The Chrome Content Security Policy](https://developer.chrome.com/docs/privacy-security/csp) doesn't allow the use of `eval` and `new Function`. 

This prevents people from building Chrome extensions that interact with Playground.

Fixes #1641

## Implementation details

[The `AJV` library doesn't work with CSP](https://github.com/ajv-validator/ajv/issues/406), so we are replacing it with `jsonschema`  for blueprint validations which does.

`jsonschema` has a slightly different error output, so we had to also adjust tests.

## Testing Instructions (or ideally a Blueprint)

- Checkout this branch
- [Open this URL](http://127.0.0.1:5400/website-server/#{%22landingPag%22:%22/wp-admin/%22,%22steps%22:%20[%20{%20%22step%22:%20%22login%22%20}]})
- Confirm that you see a error screen
- [Open this URL](http://127.0.0.1:5400/website-server/#{%22landingPage%22:%22/wp-admin/%22,%22steps%22:%20[%20{%20%22step%22:%20%22login%22%20}]})
- Confirm that WP-admin loads
